backend/manager.go

```diff
@@ -73,7 +73,7 @@ func (bm *manager) GetBackend(backendType string) (Backend, error) {
 
 	bm.wg.Add(1)
 	go func() {
-		be.Run(bm.ctx)
+		<-bm.ctx.Done()
```

Comment:
I think it would be better to do this under a new PR, but thanks for spotting it.